design/design-doc-stacks.md

```diff
@@ -195,39 +197,48 @@ spec:
 
+More details will be provided when a Stack registry project is bootstrapped and launched.
+
 Inside of a package, the following filesystem layout is expected:
 
 ```
 .registry/
-├── icon.jpg
+├── icon.svg
 ├── app.yaml # Application metadata.
 ├── install.yaml # Optional install metadata.
 ├── rbac.yaml # Optional RBAC permissions.
 └── resources
       └── databases.foocompany.io # Group directory
             ├── group.yaml # Optional Group metadata
-            ├── icon.jpg # Optional Group icon
+            ├── icon.svg # Optional Group icon
             └── mysql # Kind directory by convention
                 ├── v1alpha1
                 │   ├── mysql.v1alpha1.crd.yaml # Required CRD
-                │   ├── icon.jpg # Optional resource icon
+                │   ├── icon.svg # Optional resource icon
                 │   └── resource.yaml # Resource level metadata.
                 └── v1beta1
                     ├── mysql.v1beta1.crd.yaml
-                    ├── icon.jpg
+                    ├── icon.svg
                     └── resource.yaml
 ```
 
 * `app.yaml`: This file is the general metadata and information about the Stack, such as its name, description, version, owners, etc.  This metadata will be saved in the `Extension` record's spec fields.
 * `install.yaml`: This file contains the information for how the custom controller for the Stack should be installed into Crossplane.  Initially, only simple `Deployment` based controllers will be supported, but eventually other types of implementations will be supported as well, e.g., templates, functions/hooks, templates, a new DSL, etc.
-* `resources` directory: This directory contains all the CRDs and optional metadata about them.  These CRDs are the types that the custom controller implements the logic for.  They will be directly installed into Crossplane so that users can create instances of them to start consuming their new Stack functionality.
+* `icon.svg`: This file (or `icon.png`, `icon.jpg`, `icon.gif`, or potentially other supported filenames, TBD) will be used in a visual context when listing or describing this stack.  The preferred formats/filenames are `svg`, `png`, `jpg`, `gif` in that order (if multiple files exist).  For bitmap formats, the width to height ratio should be 1:1. Limitations may be placed on the acceptable file dimensions and byte size (TBD).
+* `resources` directory: This directory contains all the CRDs and optional metadata about them.
+  * `group.yaml`: Related Stack resources (CRDs) can be described at a high level within a group directory using this file.
+  * `*crd.yaml`: These CRDs are the types that the custom controller implements the logic for.  They will be directly installed into Crossplane so that users can create instances of them to start consuming their new Stack functionality.
```

Comment:
How did we choose `*crd.yaml` as the pattern? In working on #572 (with kubebuilder2) I'm finding it a bit inconvenient to bundle files with this naming suffix, unless I want to write a fancy tool to do it for me (which I'm trying to avoid for now). Or unless I just roll everything into a single file, which works and is probably fine but which feels less ideal. In this case, the reason it's inconvenient is: kubebuilder2 doesn't use a `crd.yaml` suffix for its crd files; I haven't found a way to control that via `controller-gen` yet; and its standard use-case is to output all files at once to an individual directory (or to `stdout`).

This is early thinking, but I am imagining these are some of our different options:
* From the stack manager, print all of the yamls in the bundle, and only create CRDs out of things which are CRDs (by inspecting the `Kind`).
* Recommend using a single file for all crds for this case; in this situation, it'd be easy to print everything to `stdout` and redirect it to a `mystuff.crd.yaml` or similar.
* Use a folder pattern (like `crds`) instead of a file pattern

I am leaning toward the stack manager change at this point, because then the bundle format is a bit more flexible, but I can see nice things about the other options too. What are your thoughts?

Reply:
The convention that I noticed in some of the existing design docs and examples was to repeat the name of the group before suffixing it with `.crd.yaml`.  I also wondered why we needed to repeat the name of the group in the filename.  That said, `*.crd.yaml` permits use of `crd.yaml` and a Stack Developer could combine all CRDs into this single file.  

In this way, I see the design as flexible.

Reply:
I've noted the arbitrary naming of the CRDs and their directory structure in the latest squash.  I'm resolving this conversation but feel free to reopen it if I overlooked some of your thinking.